### PR TITLE
Prevent connection state monitor from being cancelled on disconnect

### LINKED ISSUE #226: Can't reconnect after disconnect
Hi

On iOS when I call disconnect() the state changes directly to Disconnected, if I Connect again the state changes to Connected but on write if get the com.juul.kable.NotReadyException because the peripheral is disconnected

```
Debug: (Kermit) New state Disconnected(status=null)
Debug: (Kermit) New state com.juul.kable.State.Connecting.Bluetooth@83761768
Debug: (Kermit) New state com.juul.kable.State.Connecting.Services@83761528
Debug: (Kermit) New state com.juul.kable.State.Connected@83761718```

Now I call disconnect() the state never updates, if I call connect() again it updates directly to Connected

Debug: (Kermit) New state com.juul.kable.State.Connected@83761718

Now I write to the peripheral but I get the exception because the peripheral is disconnected.

com.juul.kable.NotReadyException

### PR DESCRIPTION
**Pull request by twyatt**
Added a dedicated connection `CoroutineScope`, so that in-flight connection attempts can be cancelled (via `Job.cancelChildren`) without cancelling the Coroutine that pipes `CentralManager` delegate callbacks to the `Peripheral`'s connection state `Flow`.

Closes #226

### CODE REVIEW

**Review by cedrickcooke — APPROVED** [review]
Just checking. Should this change be mirrored to (or is it already present in) the other targets?

**Comment by twyatt** [comment]
> Just checking. Should this change be mirrored to (or is it already present in) the other targets?

Good callout.

On Android and Javascript, we update the `StateFlow` directly from the callbacks/listeners.
From what I vaguely remember, this was not done on Apple due to freezing issues. So, something to revisit after #237.

tl;dr this change is not needed on the other platforms, thanks for double checking though!

**Review by sdonn3 — APPROVED** [review]

**Review by davidtaylor-juul — APPROVED** [review]
